server/app/controllers/likes_controller.rb

```diff
@@ -1,5 +1,49 @@
 # frozen_string_literal: true
 
 class LikesController < ApplicationController
-  def index; end
+  before_action :authenticate_user!, only: %i[create destroy]
+
+  def index
+    like = advert.likes.where(user_id: current_user.try(:id)).first 
+    render json: { message: like.try(:id), amount: advert.likes.size }
+  end
+
+  def create
+    like = advert.likes.new(user_id: current_user.id)
+    if like.valid?
+      like.save
```

Comment:
method save does trigger method `valid?`, so that it could be just `if like.save`